File: Aula_X_Recursão/loop.py
# Exibir os números pares de 1 a 100

MAX_NUMERO = 950

def loop_v1():
	for i in range(MAX_NUMERO):
		if i % 2 == 0:
			print(i, "é par!")

def loop_v2():
	for i in range(0, MAX_NUMERO, 2):
		print(i, "é par!")

def rec_v1(n):
	if n > MAX_NUMERO:
		return
	
	if n % 2 == 0:
		print(n, "é par!")

	rec_v1(n + 1)

def rec_v2(n):
	if n > MAX_NUMERO:
		return

	print(n, "é par!")
	rec_v2(n + 2)


if __name__ == "__main__":
	# print("loop_v1")
	# loop_v1()
	
	
	"""
	print("loop_v2")
	loop_v2()
	"""

	rec_v1(0)
	print("rec_v1")
	
	"""
	rec_v2(0)
	print("rec_v2")
	"""
# loop_v1 = 31s
# loop_v1 = 36s
# rec_v1 = 

File: Aula_X_Recursão/loop.c
#include <stdio.h>

// Exibir os numeros pares de 1 a MAX_NUMERO

int MAX_NUMERO = 500000;

void loop_v1() {
	for (int i = 0; i <= MAX_NUMERO; i++) {
		if (i % 2 == 0) {
			// printf("%d é par\n", i);
		}
	}
}

void loop_v2() {
	for (int i = 0; i <= MAX_NUMERO; i += 2) {
		// printf("%d é par\n", i);
	}
}

void rec_v1(int n) {
	if (n > MAX_NUMERO) {
		return;
	}

	if (n % 2 == 0) {
		// printf("%d é par!\n", n);
	}

	rec_v1(n + 1);
}

void rec_v2(int n) {
	if (n > MAX_NUMERO) {
		return;
	}

	// printf("%d é par!\n", n);

	rec_v2(n + 2);
}

int main() {
	
    //loop_v1();
	//printf("loop_v1\n");

	// loop_v2();
	// printf("loop_v2\n");

	// rec_v1(0);
	// printf("rec_v1\n");

	rec_v2(0);
	//printf("rec_v2\n");

	return 0;
}